js/game.js

```diff
@@ -207,10 +208,46 @@ function rivalBet() {
             game.rivalBet = game.lastBetStanding;
     }
 };
+
 // Results step
-function showResults() {
+function startResults() {
     game.step = Steps.RESULTS;
-    game.renderFun();
+    calculateResults(Subgames.BIGGEST);
+};
+
+function calculateResults(subgame) {
+    game.subgame = subgame;
+    let pendingBet = game.pendingBets[subgame];
+    if (pendingBet) {
+        if (determineWinner(subgame, game.playerDice, game.rival.dice, game.playerStartsBets) == 1) {
+            game.playerPoints += pendingBet;
+        } else {
+            game.rivalPoints += pendingBet;
+        }
+    }
+    game.selectedDiceKeys.clear();
+    let playerBestDice = bestDice(subgame, game.playerDice);
+    playerBestDice.forEach(x => game.selectedDiceKeys.add(x));
+    game.renderFun(game);
+};
+
+function nextResults() {
+    let nextSubgame = game.subgame + 1;
+    if (nextSubgame < Object.keys(Subgames).length) {
+        calculateResults(nextSubgame);
+    } else {
+        // TODO: Add "reflection" step
+        nextRoundOrFinish();
+    }
+}
+
+function nextRoundOrFinish() {
+    if (game.playerPoints >= 30 || game.rivalPoints >= 30 && !(game.playerPoints == game.rivalPoints)) {
+        game.step = Steps.FINISH;
+    } else {
+        //TODO: game.playerStartsBets = !game.playerStartsBets;
+        rollInitialDice();
```

Comment:
maybe a reflection step here as well, before the next round? if you want, a comment is enough for now so we don't forget

Reply:
I've already left a comment on the only caller of this function `nextResults()`, so it should be ok if it's implemented here or there.